core/src/test/java/org/elasticsearch/http/netty/NettyHttpChannelTests.java

```diff
@@ -414,5 +460,68 @@ public BytesReference content() {
         public RestStatus status() {
             return RestStatus.OK;
         }
+
+    }
+
+    private static class TestCircuitBreaker implements CircuitBreaker {
```

Comment:
I think we need to make all the methods synchronized here? we don't have any concurrency guarantees right now